daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Blinding.scala

```diff
@@ -47,16 +52,12 @@ object Blinding {
         }
         .mkString(";")
     }
-    if (enrichedTx.failedAuthorizations.isEmpty) {
-      Right(
-        BlindingInfo(
-          disclosure = enrichedTx.explicitDisclosure,
-          divulgence = enrichedTx.implicitDisclosure,
-        ))
+    if (failedAuthorizations.isEmpty) {
+      Right(blindingInfo)
```

Comment:
Wouldn't it make more sense to calculate blinding only if we get here instead of doing it anyway?

Reply:
Yes it would. Thanks.

Reply:
Someone's coming from lazy-land. :wink: